engine/dlib/src/dlib/sys_posix.cpp

```diff
@@ -597,6 +598,103 @@ namespace dmSys
 #endif
 
 
+#if defined(__EMSCRIPTEN__)
+    Result GetHomePath(char* path, uint32_t path_len)
+    {
+        return GetApplicationSupportPath("", path, path_len);
+    }
+#elif defined(__ANDROID__)
+    Result GetHomePath(char* path, uint32_t path_len)
+    {
+        dmAndroid::ThreadAttacher thread;
+        JNIEnv* env = thread.GetEnv();
+        if (!env)
+        {
+            return RESULT_UNKNOWN;
+        }
+
+        jclass activity_class = env->FindClass("android/app/NativeActivity");
+        jobject activity_obj = thread.GetActivity()->clazz;
+
+        jstring path_obj = NULL;
+
+        // First, try getDataDir() (API 24+)
+        jmethodID get_data_dir_method = env->GetMethodID(activity_class, "getDataDir", "()Ljava/io/File;");
+        if (get_data_dir_method) {
+            jobject data_dir_obj = env->CallObjectMethod(activity_obj, get_data_dir_method);
+            if (data_dir_obj) {
+                jclass file_class = env->FindClass("java/io/File");
+                jmethodID getPathMethod = env->GetMethodID(file_class, "getPath", "()Ljava/lang/String;");
+                path_obj = (jstring) env->CallObjectMethod(data_dir_obj, getPathMethod);
+            }
+        }
```

Comment:
This is a preferable way to do it, once we update minimal android version to 24 we will keep only this one